src/utils/reqwest_resume/mod.rs

```diff
@@ -0,0 +1,165 @@
+                } else {
+                    false
+                };
+            let resp = Response {
+                _client: client,
+                _method: method,
+                _url: url,
+                _response: response,
+                _accept_byte_ranges: accept_byte_ranges,
+                _pos: 0,
+            };
+            Ok(resp)
+        }
+    }
+}
+
+/// A Response to a submitted Request.
+///
+/// See [`reqwest::Response`].
+#[derive(Debug)]
+pub struct Response {
+    _client: reqwest::Client,
+    _method: reqwest::Method,
+    _url: reqwest::Url,
+    _response: reqwest::Response,
+    _accept_byte_ranges: bool,
+    _pos: u64,
+}
+
+impl Response {
+    pub fn response(self) -> reqwest::Response {
+        self._response
+    }
+}
+
+struct Decoder {
+    client: reqwest::Client,
+    method: reqwest::Method,
+    url: reqwest::Url,
+    decoder: Pin<Box<dyn Stream<Item = reqwest::Result<Bytes>> + Send + Unpin>>,
+    accept_byte_ranges: bool,
+    pos: u64,
+}
+impl Stream for Decoder {
+    type Item = reqwest::Result<Bytes>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
+        loop {
+            match ready!(self.decoder.as_mut().poll_next(cx)) {
+                Some(Err(err)) => {
+                    if !self.accept_byte_ranges {
+                        // TODO: we could try, for those servers that don't output Accept-Ranges but work anyway
+                        break Poll::Ready(Some(Err(err)));
+                    }
+                    let builder = self.client.request(self.method.clone(), self.url.clone());
+                    let mut headers = HeaderMap::new();
+                    let value = HeaderValue::from_str(&std::format!("bytes={}-", self.pos))
+                        .expect("invalid ASCII string");
+                    headers.insert(header::RANGE, value);
+                    let builder = builder.headers(headers);
+                    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests
+                    // https://github.com/sdroege/gst-plugin-rs/blob/dcb36832329fde0113a41b80ebdb5efd28ead68d/gst-plugin-http/src/httpsrc.rs
+                    self.decoder = Box::pin(
+                        Box::pin(sleep(Duration::from_secs(1)))
+                            .then(|()| builder.send())
+                            .map_ok(reqwest::Response::bytes_stream)
+                            .try_flatten_stream(),
+                    );
+                }
+                Some(Ok(n)) => {
+                    self.pos += n.len() as u64;
+                    break Poll::Ready(Some(Ok(n)));
+                }
+                None => break Poll::Ready(None),
+            }
+        }
+    }
+}
+
+/// Shortcut method to quickly make a GET request.
+///
+/// See [`reqwest::get`].
+pub fn get(url: reqwest::Url) -> impl Future<Output = reqwest::Result<Response>> + Send {
+    // <T: IntoUrl>
```

Comment:
Do we need this comment?

Reply:
Not really. Keep in mind that most of the code comes from the `reqwest_resume` crate and I did minimal modifications.

Reply:
I understand that. Let's remove everything that seems redundant, unless there is a good reason not to.